src/components/pages/PageDecksEdit/PageDecksEdit.test.jsx

```diff
@@ -0,0 +1,188 @@
+import '../../../tests/mocks/template-default-mocks';
+import '../../../tests/mocks/firebase-mocks';
+import React from 'react';
+import { mount } from 'enzyme';
+import { MemoryRouter } from 'react-router';
+import configureMockStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import { act } from 'react-dom/test-utils';
+import PageDecksEdit from './PageDecksEdit';
+import { findByTestAttr } from '../../../tests/testUtils';
+import firebaseCollectionsHelper from '../../../firebase/firebase-collections-helper/firebase-collections-helper';
+
+const defaultArgs = {
+  user: {},
+  cards: [],
+  deck: {},
+  cardId: '',
+  deckCardRef: '',
+};
+
+describe('PageDecksEdit', () => {
+  const setup = async (args = {}) => {
+    const {
+      user, cards, deck, cardId,
+    } = {
+      ...defaultArgs,
+      ...args,
+    };
+
+    const mockStore = configureMockStore([]);
+    const store = mockStore({ user });
+
+    const spyGetCardOptions = await jest
+      .spyOn(firebaseCollectionsHelper, 'getAllCardsDataByType');
+    spyGetCardOptions.mockReturnValue(Promise.resolve(cards));
+
+    const spyPopulateData = await jest
+      .spyOn(firebaseCollectionsHelper, 'getDeckData');
+
+    spyPopulateData.mockReturnValue(Promise.resolve(deck));
+
+    const spyGetCardRef = jest
+      .spyOn(firebaseCollectionsHelper, 'getCardRef');
+
+    spyGetCardRef.mockReturnValue(Promise.resolve(cardId));
+
+    const spyHandleUpdate = jest
+      .spyOn(firebaseCollectionsHelper, 'updateDeck');
+
+    spyHandleUpdate.mockReturnValue(Promise.resolve(deck));
+
+    let wrapper;
+    await act(async () => {
+      wrapper = await mount(
+        <Provider store={store}>
+          <MemoryRouter>
+            <PageDecksEdit />
+          </MemoryRouter>
+        </Provider>,
+      );
+    });
+
+    return {
+      wrapper,
+      spyGetCardOptions,
+      spyPopulateData,
+      spyGetCardRef,
+      spyHandleUpdate,
+    };
+  };
+
+  it('should render without error', async () => {
+    const deck = {
```

Comment:
You repeat creating this deck several times. Consider moving the deck creation to a function.